pontoon/base/models.py

```diff
@@ -472,6 +507,93 @@ def cldr_id_to_plural(self, cldr_id):
     def nplurals(self):
         return len(self.cldr_plurals_list())
 
+    @property
+    def projects_permissions(self):
+        """
+        List of tuples that contain informations required by the locale permissions view.
+
+        A row contains:
+            id  - id of project locale
+            slug - slug of the project
+            name - name of the project
+            all_users - (id, first_name, email) users that aren't translators of given project locale.
+            translators - (id, first_name, email) translators assigned to a project locale.
+            contributors - (id, first_name, email) people who contributed for a project in current locale.
+            has_custom_translators - Flag if locale has a translators group for the project.
+        """
+        locale_projects = []
+
+        def create_users_map(qs, keyfield):
+            """
+            Creates a map from query set
+            """
+            user_map = defaultdict(list)
+            for row in qs:
+                key = row.pop(keyfield)
+                user_map[key].append(row)
+            return user_map
+
+        project_locales = list(
+            self.project_locale.all()
+                .prefetch_related('project', 'translators_group')
+                .values(
+                    'id',
+                    'project__pk',
+                    'project__name',
+                    'project__slug',
+                    'translators_group__pk',
+                    'has_custom_translators'
+                )
+        )
+
+        projects_translators = create_users_map(
+            User.objects
+                .filter(groups__pk__in=[project_locale['translators_group__pk'] for project_locale in project_locales])
+                .exclude(email='')
+                .prefetch_related('groups')
+                .values('id', 'first_name', 'email', 'groups__pk')
+                .distinct(),
+            'groups__pk'
+        )
+
+        projects_contributors = create_users_map(
+            User.objects
+                .filter(
+                    translation__locale=self,
+                    translation__entity__resource__project__pk__in=[project_locale['project__pk'] for project_locale in project_locales]
+                )
+                .exclude(email='')
+                .prefetch_related('translation__entity__resource__project')
+                .values('id', 'first_name', 'email', 'translation__entity__resource__project__pk')
+                .distinct(),
+            'translation__entity__resource__project__pk'
+        )
+
+        projects_all_users = defaultdict(set)
```

Comment:
This is actually all users without `projects_translators`. Shall we simply use exclude?
https://github.com/mozilla/pontoon/blob/master/pontoon/base/views.py#L129